mesonbuild/compilers/compilers.py

```diff
@@ -548,6 +548,22 @@ def to_native(self, copy: bool = False) -> typing.List[str]:
                 # Last occurrence of a library
                 new.insert(group_end + 1, '-Wl,--end-group')
                 new.insert(group_start, '-Wl,--start-group')
+        # Remove system/default include paths added with -isystem
+        if hasattr(self.compiler, 'get_default_include_dirs'):
+            default_dirs = self.compiler.get_default_include_dirs()
+            bad_idx_list = []
+            for i, each in enumerate(new):
+                # Remove the current and and next list element
```

Comment:
This comment says what the code does, which also is obvious from reading the code. Please document *why*.